lib/cuke_linter/linters/unique_scenario_names_linter.rb

```diff
@@ -0,0 +1,107 @@
+module CukeLinter
+  # A linter that detects non-unique scenario names
+  class UniqueScenarioNamesLinter < Linter
+
+    def initialize
+      super
+      @scenario_names = {}
+    end
+
+    def rule(model)
+      return nil unless valid_model?(model)
+
+      feature_file = model.get_ancestor(:feature_file)
+      return nil if feature_file.nil?
+
+      file_path = feature_file.path
+
+      case model
+      when CukeModeler::Rule
+        check_rule(model, file_path)
+      when CukeModeler::Scenario, CukeModeler::Outline
+        return nil if model.get_ancestor(:rule)
+
+        scope_key = "#{file_path}:feature"
+
+        if model.is_a?(CukeModeler::Outline)
+          check_scenario_outline(model, scope_key)
+        else
+          check_scenario(model, scope_key)
+        end
+      else
+        nil
+      end
```

Comment:
Something like that:

```
case model
      when CukeModeler::Rule
        check_rule(model, file_path)
      when CukeModeler::Scenario, CukeModeler::Outline
        return nil if model.get_ancestor(:rule)

        scope_key = "#{file_path}:feature"
        check_scenario_or_outline(model, scope_key)
      end
```

and new method:

```
   def check_scenario_or_outline(model, scope_key)
      if model.is_a?(CukeModeler::Outline)
        check_scenario_outline(model, scope_key)
      else
        check_scenario(model, scope_key)
      end
    end
    ```

Reply:
Done